snuba/consumers/consumer.py

```diff
@@ -601,15 +608,92 @@ class MultistorageKafkaPayload(NamedTuple):
+
+    # expected format is "[topic:partition_index:offset,...]" eg [snuba-metrics:0:1,snuba-metrics:0:3]
+    messages_to_skip = (get_config("kafka_messages_to_skip") or "[]")[1:-1].split(",")
+    return (
+        f"{message.partition.topic.name}:{message.partition.index}:{message.offset}"
+        in messages_to_skip
+    )
+
+
 def process_message(
-    processor: MessageProcessor, message: Message[KafkaPayload]
+    processor: MessageProcessor, consumer_group: str, message: Message[KafkaPayload]
 ) -> Union[None, BytesInsertBatch, ReplacementBatch]:
-    result = processor.process_message(
-        rapidjson.loads(message.payload.value),
-        KafkaMessageMetadata(
-            message.offset, message.partition.index, message.timestamp
-        ),
-    )
+
+    # TODO: Remove temporary mitigations for DLQ prod test
+
+    # Mitigation: Last resort - Hardcode message(s) to skip in runtime config
+    if skip_kafka_message(message):
+        logger.warning(
+            f"A consumer for {message.partition.topic.name} skipped a message!",
+            extra=__message_to_dict(message),
+        )
+        return None
+
+    # Mitigation: JSON fails to deserialize
+    try:
+        deserialized_message = rapidjson.loads(message.payload.value)
+    except Exception as err:
+        if message.partition.topic.name == StreamsTopic.METRICS.value:
+            logger.error(
+                err,
+                exc_info=True,
+                extra=__message_to_dict(message),
+            )
+            return None
+        raise err
+
+    try:
+        result = processor.process_message(
+            deserialized_message,
+            KafkaMessageMetadata(
+                message.offset, message.partition.index, message.timestamp
+            ),
+        )
+    except Exception as err:
+        # TODO: Remove if condition, leaving only code in else block after DLQ prod test
+        if (
+            message.partition.topic.name == StreamsTopic.METRICS.value
+            and not get_config("enable_metrics_dlq", False)
+        ):
+            logger.error(
+                err,
+                exc_info=True,
+                extra={"cause": "Caught an exception on Metrics! (DLQ was disabled)"},
+            )
+            return None
+        else:
+            logger.error(err, exc_info=True)
+            raise InvalidMessages(
+                [
+                    InvalidKafkaMessage(
+                        payload=message.payload.value,
+                        timestamp=message.timestamp,
+                        topic=message.partition.topic.name,
+                        consumer_group=consumer_group,
+                        partition=message.partition.index,
+                        offset=message.offset,
+                        headers=message.payload.headers,
+                        key=message.payload.key,
+                        reason=str(err.args),
+                    )
+                ]
+            ) from err
```

Comment:
This code block will remain in the error handling of `process_message`.

Adding `logger.error()` like I have in line 681 ensures that the exception is logged to Sentry (in case the DLQ handles it), but it doesn't really log the real exception which will be `InvalidMessages`.

I think ff there isn't a DLQ to handle it this would be double logged (once by logger.error() and once by raising the unhandled exception itself). Maybe the DLQ try/except itself should be logging to Sentry instead?